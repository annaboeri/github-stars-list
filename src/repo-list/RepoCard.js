import React, { useState } from 'react';
import CommitListContainer from './commit-list/CommitListContainer';
import './RepoListContainer.css';

const RepoCard = props => {
  const [showCommitList, setShowCommitList] = useState(false);
  const { repo } = props;

  return (
    <li className={`flip-card ${showCommitList ? 'flip' : ''}`}>
      <div className="card-content">
        <div className={`flip-card-front ${showCommitList ? 'hide' : ''}`}>
          <h3>{repo.name}</h3>
          <p className="ellipsis">{repo.description}</p>
          <a href={repo.html_url} className="repo-link">
            <i className="fab fa-github"></i>
            View Repo
          </a>
          <p>
            <i className="fas fa-star"></i>{repo.stargazers_count.toLocaleString()}
            <i className="fas fa-code-branch"></i>{repo.forks_count.toLocaleString()}
          </p>
          <button
            className="view-commits-btn"
            onClick={(prevState) => setShowCommitList(!showCommitList)}
          >
            View Recent Commmits
          </button>
        </div>
        <div className="flip-card-back">
          <button
            className="back-btn"
            onClick={(prevState) => setShowCommitList(!showCommitList)}
          >
            <i className="fas fa-arrow-left"></i> Back
          </button>
            { showCommitList && (
              <CommitListContainer
                repoName={repo.full_name}
                handleErrors={props.handleErrors}
              />
            )}
        </div>
      </div>
    </li>
  )
}

export default RepoCard;
