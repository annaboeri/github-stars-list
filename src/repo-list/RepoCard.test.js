import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import RepoCard from './RepoCard';

describe('RepoCard Component', () => {
  const props = {
    repo: {
      name: 'My Repo Name',
      desc: 'My Repo Desc',
      html_url: '/',
      stargazers_count: 1,
      forks_count: 2,
      full_name: 'full/name',
    },
    handleErrors: jest.fn(),
  }

  it('should render correctly', (done) => {
    const component = shallow(<RepoCard {...props} />);
    const tree = toJson(component);

    expect(tree).toMatchSnapshot();
    expect(component.find('.flip-card').exists()).toBe(true);
    done();
  });
});
