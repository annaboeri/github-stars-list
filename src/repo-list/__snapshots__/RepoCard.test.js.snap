// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoCard Component should render correctly 1`] = `
<li
  className="flip-card "
>
  <div
    className="card-content"
  >
    <div
      className="flip-card-front "
    >
      <h3>
        My Repo Name
      </h3>
      <p
        className="ellipsis"
      />
      <a
        className="repo-link"
        href="/"
      >
        <i
          className="fab fa-github"
        />
        View Repo
      </a>
      <p>
        <i
          className="fas fa-star"
        />
        1
        <i
          className="fas fa-code-branch"
        />
        2
      </p>
      <button
        className="view-commits-btn"
        onClick={[Function]}
      >
        View Recent Commmits
      </button>
    </div>
    <div
      className="flip-card-back"
    >
      <button
        className="back-btn"
        onClick={[Function]}
      >
        <i
          className="fas fa-arrow-left"
        />
         Back
      </button>
    </div>
  </div>
</li>
`;
